react/private/withTextProps.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import omit from 'lodash/omit';
+
+export const sizes = [
+  'small',
+  'substandard',
+  'superstandard',
+  'standard',
+  'subheading',
+  'headline',
+  'heading',
+  'hero'
+];
+
+export const SizePropTypes = PropTypes.oneOf(sizes);
```

Comment:
I think this should probably be an object that contains the boolean props too, which then gets spread into other proptypes objects, maybe renamed to something like `withTextPropTypes`, e.g.

```js
Text.propTypes = {
  ...withTextPropTypes
};
```